### PR TITLE
[20411] Improve junit_summary robustness

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

#56 introduced a new action to create summaries from jUnit reports. It also introduced support in the colcon_test action to create jUnit reports from CTest ones for CMake versions where CTest does not have native support for jUnit reporting. For the latter cases, the generated jUnit reports where not complete enough, and the junit_summary.py script was not robust enough to cope with the differences. This PR

* Extends the generated reports to have some missing required properties
* Improves the robustness to the junit_summary.py script so if can generate summaries from the different jUnit reports.

Depends on:
* #57 

## Contributor Checklist
- [x] Commit messages follow the company guidelines. 
- _N/A_: Changes are API compatible. 
- _N/A_: New features have been added to the `versions.md` and `README.md` files (if applicable).

## Reviewer Checklist
- [x] The title and description correctly express the PR's purpose.
- [x] The Contributor checklist is correctly filled.

### CODE REVIEW

File: multiplatform/colcon_test/action.yml [hunk 1]
```diff
@@ -37,6 +37,13 @@ inputs:
     required: false
     default: ''
 
+  test_report_artifact:
+    description: >
+      Name of the artifact to be uploaded containing the test report in JUnit format. It shall be unique withing the workflow run.
```
Comment (MiguelCompany) [inline multiplatform/colcon_test/action.yml:42]:
```suggestion
      Name of the artifact to be uploaded containing the test report in JUnit format. It shall be unique within the workflow run.
```

File: resources/junit_summary.py [hunk 2]
```diff
@@ -77,34 +77,67 @@ def parse_options():
...
 
             elif child.attrib['status'] == "notrun":
-                result['skipped_tests'].append(child.attrib['name'])
+                suite_result['skipped_tests'].append(child.attrib['name'])
+
+    # Update result
+    suite_name = 'Not specified'
+    if 'name' in test_suite.attrib and test_suite.attrib['name'] != '(empty)':
+        suite_name = test_suite.attrib['name']
+    result[suite_name] = suite_result
```
Comment (MiguelCompany) [inline resources/junit_summary.py:140]: This could override results with no name

File: ubuntu/colcon_test/action.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ inputs:
 
   test_report_artifact:
     description: >
-      Name of the artifact to be uploaded containing the test report in JUnit format.
+      Name of the artifact to be uploaded containing the test report in JUnit format. It shall be unique withing the workflow run.
```
Comment (MiguelCompany) [inline ubuntu/colcon_test/action.yml:42]:
```suggestion
      Name of the artifact to be uploaded containing the test report in JUnit format. It shall be unique within the workflow run.
```

File: windows/colcon_test/action.yml [hunk 4]
```diff
@@ -39,7 +39,7 @@ inputs:
 
   test_report_artifact:
     description: >
-      Name of the artifact to be uploaded containing the test report in JUnit format
+      Name of the artifact to be uploaded containing the test report in JUnit format. It shall be unique withing the workflow run.
```
Comment (MiguelCompany) [inline windows/colcon_test/action.yml:42]:
```suggestion
      Name of the artifact to be uploaded containing the test report in JUnit format. It shall be unique within the workflow run.
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: resources/junit_summary.py [hunk 5]
```diff
@@ -77,34 +77,67 @@ def parse_options():
...
+    # Root can be 'testsuites' or 'testsuite'
+    if root.tag == 'testsuites':
+        for test_suite in root:
+            result = parse_testsuite(test_suite, result)
+    elif root.tag == 'testsuite':
+        print('parsing testsuite')
+        result = parse_testsuite(root, result)
+
+    print(result)
+    return result
```
Comment (MiguelCompany) [inline resources/junit_summary.py:95]: Debug prints?
```suggestion
    result = {}
    tree = ET.parse(junit_report)
    root = tree.getroot()

    # Root can be 'testsuites' or 'testsuite'
    if root.tag == 'testsuites':
        for test_suite in root:
            result = parse_testsuite(test_suite, result)
    elif root.tag == 'testsuite':
        result = parse_testsuite(root, result)

    return result
```

File: resources/junit_summary.py [hunk 6]
```diff
@@ -77,72 +77,120 @@ def parse_options():
...
+        summary += f'|{suite["skipped"]}'
+        summary += f'|{suite["time"]}'
+        summary += f'|{suite["timestamp"]}'
+        summary += '|\n'
+
+    # Test lists
+    for suite in results:
+        # Failed tests list
+        if show_failed is True and len(suite['failed_tests']) != 0:
+            summary += f'\n### Failed tests in suite: {suite["name"]}\n'
```
Comment (MiguelCompany) [inline resources/junit_summary.py:173]:
```suggestion
            summary += f'\n### Suite `{suite["name"]}` had {len(suite['failed_tests'])} failed tests:\n'
```

Comment (MiguelCompany) [reply resources/junit_summary.py:173]: And similar changes for disabled and skipped

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Comment by EduPonz** [comment]
Tested in:

* eProsima/Fast-DDS#4361 [run](https://github.com/eProsima/Fast-DDS/actions/runs/7900126837?pr=4361)

**Pull request merged**